index/secondary_index.cc

```diff
@@ -106,6 +106,26 @@ sstring target_parser::get_target_column_name_from_string(const sstring& targets
     return targets;
 }
 
+sstring target_parser::get_base_column_name_from_target(const sstring& target) {
+    sstring base_name = target;
+    
+    // extract column name if the target is in format "function(column_name)"
+    auto opening_pos = target.find("(");
+    if (opening_pos != target.npos) {
+        auto closing_pos = target.find(")", ++opening_pos);
+        if (opening_pos != target.npos) {
+            base_name = target.substr(opening_pos, closing_pos - opening_pos);
+        }
+    }
+
+    // unquote column name if it's quoted
+    if (base_name[0] == '"' && base_name[base_name.length() - 1] == '"') {
```

Comment:
What if `base_name` is empty? Not sure if it can happen at all, but I would feel more confident about this code if this case were properly handled. This also doesn't seem to work if the string is just a single `"` (you will try to call substr with negative length).

Reply:
If the string might be quoted then the next natural question is, "can this string contain some escaped characters"?

Which code/function is responsible for adding the quotes? Is there some unquote logic/function that exists and could be used here?

Reply:
> Which code/function is responsible for adding the quotes? Is there some unquote logic/function that exists and could be used here?

`cql3::util::maybe_quote()` and it contains logic to replace `"` in the name with `""`. But AFAIK it cannot contain other escaped characters.
@nyh you are the expert of cql

Reply:
Yes, we have the function index_target::unescape_target_column() and you should use it.
It was introduced in https://github.com/scylladb/scylladb/pull/10841/commits/2c244c6e09c4693a47eadded930651599f8145d1 which also explains why this quoting was needed.